tests/api/controllers/robot_parallel_step/robot_parallel_step.c

```diff
@@ -74,29 +63,41 @@ int main(int argc, char **argv) {
 
   wb_robot_step_end();  // retrieve sensor values after step update
 
-  // step 3 - 96 ms -> check camera values after first pose update (second is only visible by the sensor after the next
+  // step 3 - 96 ms -> check sensor values after first pose update (second is only visible by the sensor after the next
   // step_end)
   wb_robot_step_begin(time_step);
 
-  image = wb_camera_get_image(camera);
+  // compass
+  values = wb_compass_get_values(compass);
+  const double newCompassValues1[3] = {0.0, 0.0, 1.0};
+  ts_assert_doubles_in_delta(3, values, newCompassValues1, 0.0001, "Compass measurement after first pose update is wrong.");
+
+  // distance sensor
+  distance = wb_distance_sensor_get_value(ds);
+  ts_assert_double_in_delta(distance, 650.0, 0.001, "Distance sensor measurement after first pose update is wrong.");
 
-  // intense computation between begin and end
-  for (int i = 0; i < 1000000; i++) {
-    r = wb_camera_image_get_red(image, 1, 0, 0);
-    g = wb_camera_image_get_blue(image, 1, 0, 0);
-    b = wb_camera_image_get_green(image, 1, 0, 0);
-  }
```

Comment:
We should keep this loop (or replace it by something similar) even if the camera values won't be used because it is an important part of the test.
Parallelization only makes sense if intense computation occurs on the controller side and that is the main case we should test.

Reply:
I've added some computation along with some sleeping.